monkey/infection_monkey/master/propagator.py

```diff
@@ -153,6 +153,17 @@ def _process_tcp_scan_results(
             if psd.banner is not None:
                 target_host.services[psd.service_deprecated]["banner"] = psd.banner
 
+            # keep latest information about a port, but retain old information about `services`
```

Comment:
Overwrite service (singular). At some point we'll need to establish some kind of priority or precedence, but for now our fingerprinters shouldn't clobber each other since they're all checking very different things.